logstash-core-event-java/src/main/java/com/logstash/RubyJavaObject.java

```diff
@@ -0,0 +1,124 @@
+package com.logstash;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import org.jruby.Ruby;
+import org.jruby.RubyNil;
+import org.jruby.runtime.builtin.IRubyObject;
+
+import java.io.InvalidObjectException;
+import java.io.ObjectStreamException;
+import java.io.Serializable;
+import java.io.ObjectInputStream;
+
+public final class RubyJavaObject implements Serializable {
+    private static final long serialVersionUID = 4358368393114318978L;
+
+    private transient IRubyObject rubyValue;
+    private transient boolean isRubyNil;
+    private Object javaValue;
+
+    public RubyJavaObject(Object javaValue) {
+        this.javaValue = javaValue;
+        this.rubyValue = null;
+    }
+
+    public RubyJavaObject(IRubyObject rubyValue) {
+        this.rubyValue = rubyValue;
+        isRubyNil = (this.rubyValue instanceof RubyNil);
+        this.javaValue = null;
+    }
+
+    public IRubyObject getRubyValue(Ruby runtime) {
+        if (rubyValue == null) {
+            rubyValue = Rubyfier.deep(runtime, javaValue);
+        }
+        return rubyValue;
+    }
+
+    @JsonValue
+    public Object getJavaValue() {
+        if (hasRubyValue() && isRubyNil) {
+            return null;
+        }
+        if (javaValue == null) {
+            javaValue = Javafier.deep(rubyValue);
+        }
+        return javaValue;
+    }
+
+    public Object javaGetRubyValue() {
```

Comment:
I am not sure what `javaGetRubyValue` means? is it meant as returning the unconverted rubyValue ?